hack/verify-no-merge-commits.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+cd "${KUBE_ROOT}"
+
+set -x
+git show HEAD
+git log --oneline | head -n 50 || true
+
+# potentially skip final merge commit used by CI
+head=HEAD
+while [[ "$(git show -q --format="%ae" ${head})" = "k8s_bootstrap@"* ]]; do
+    if git show -q ${head} | grep -q "^Merge: "; then
+	head=$(git rev-parse ${head}^2)
```

Comment:
why `^2`?

Reply:
Second parent of a merge, i.e. "to the right" (which is usually the PR branch).

Reply:
can you maybe leave a comment explaining that?

Reply:
done